Adding presto data source

MichaelLantz opened the pull request:
Fixes mindsdb/mindsdb#1689

## Please describe what changes you made, in as much detail as possible
  - Adding datasource for Presto
  - Adding test against Presto datasource
  - Adding initialization check for Presto datasource
  - Updating dependencies / requirements for new presto python library

datasources

MichaelLantz committed:
Adding presto data source

Integrating Presto data source, ensuring the initialization checks for Presto data source and handles exception appropriately, adding test using Snowflake as a reference  test to attempt breaking data source.

MichaelLantz commented:
Still attempting to get my local tests working since all but file data sources test fail but the core of the data source has been created.

ZoranPandovski commented on mindsdb_datasources/__init__.py:
```diff
@@ -69,3 +69,9 @@
+try:
+    from mindsdb_datasources.datasources.presto_ds import PrestoDS
+except:
+    print("PrestoDS is not available by default. Please install snowflake's ODBC driver and pyodbc to use it!")
```
Please change the snowflake message

ZoranPandovski commented on mindsdb_datasources/datasources/presto_ds.py:
```diff
@@ -0,0 +1,45 @@
+            http_scheme=self.protocol
+        )
+
+        df = pd.read_sql(q, con=con)
```
Why we need pandas here?

ZoranPandovski commented on mindsdb_datasources/datasources/presto_ds.py:
```diff
@@ -0,0 +1,45 @@
+        )
+
+        df = pd.read_sql(q, con=con)
+        con.close()
```
We need to close the connection last

MichaelLantz committed:
Adding presto data source

Integrating Presto data source, ensuring the initialization checks for Presto data source and handles exception appropriately, adding test using Snowflake as a reference  test to attempt breaking data source.

ZoranPandovski committed:
Merge branch 'staging' into presto-connector

ZoranPandovski commented:
****CLA Assistant Lite bot**** All contributors have signed the CLA  ✍️

MichaelLantz commented:
I have read the CLA Document and I hereby sign the CLA
